jira-automation.py

```diff
@@ -0,0 +1,278 @@
+        os.path.isfile(rel_file_path)
+    except SystemError:
+        logging.error("the file does not exists")
+        sys.exit(1)
+
+    component_maping = {}
+    try:
+        with open(rel_file_path) as f:
+            data = f.read()
+            component_maping = json.loads(data)
+    except SystemError:
+        logging.error("failed to load file")
+        sys.exit(1)
+    return component_maping
+
+
+def main(args):
+    logging.basicConfig(level=logging.INFO)
+
+    jira_component_mapping_file_path = (
+        os.environ.get("COMPONENT_MAPPING_FILE_PATH")
+        if os.environ.get("COMPONENT_MAPPING_FILE_PATH")
+        else "./config/jira_components_mapping.json"
+    )
+    components_mapping = load_mapping(jira_component_mapping_file_path)
+
+    exclude_files_file_path = (
+        os.environ.get("EXCLUDE_FILES_FILE_PATH")
+        if os.environ.get("EXCLUDE_FILES_FILE_PATH")
+        else "./config/exclude_files.json"
+    )
+    exclude_files_mapping = load_mapping(exclude_files_file_path)
+
+    snyk_org_id = os.environ.get("SNYK_ORG_ID")
+    if not snyk_org_id:
+        logging.error("SNYK_ORG_ID env variable not defined")
+        sys.exit(2)
+    snyk_api_token = os.environ.get("SNYK_API_TOKEN")
+    if not snyk_api_token:
+        logging.error("SNYK_API_TOKEN env variable not defined")
+        sys.exit(2)
+    jira_server = os.environ.get("JIRA_SERVER")
+    if not jira_server:
+        logging.error("JIRA_SERVER env variable not defined")
+        sys.exit(2)
+    jira_api_token = os.environ.get("JIRA_API_TOKEN")
+    if not jira_api_token:
+        logging.error("JIRA_API_TOKEN env variable not defined")
+        sys.exit(2)
+    jira_project_id = os.environ.get("JIRA_PROJECT_ID")
+    if not jira_project_id:
+        logging.error("JIRA_PROJECT_ID env variable not defined")
+        sys.exit(2)
+
+    jira_label_prefix = (
+        os.environ.get("JIRA_LABEL_PREFIX")
+        if os.environ.get("JIRA_LABEL_PREFIX")
+        else "snyk-jira-integration:"
+    )
+
+    dry_run = os.environ.get("DRY_RUN")
+    if dry_run:
+        logging.info("DRY_RUN is enabled")
+
+    snyk_client = SnykClient(snyk_api_token)
+    snyk_org = snyk_client.get_organization(snyk_org_id)
+    projects = snyk_org.projects.all()
+    jira_client = JiraClient(
+        jira_server,
+        jira_api_token,
+        jira_label_prefix,
+        jira_project_id,
+        components_mapping,
+        dry_run,
+    )
+    process_projects(
+        jira_client,
+        snyk_org_id,
+        snyk_api_token,
+        projects,
+        exclude_files_mapping,
+        args.limit,
+        args.version,
+    )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Snyk to Jira automation script")
+    parser.add_argument(
+        "-l",
+        "--limit",
+        type=str,
+        help="The number of results to be returned by the snyk scan",
+        nargs="?",
+        const="100",
```

Comment:
it should be `default`, no?

Reply:
I don't know the behavior of const, but I'd guess it would override anything passed as parameters, no?

Reply:
you are right, I misunderstood const sorry

Reply:
fixed now